api/handlers.go

```diff
@@ -662,6 +798,36 @@ func (api *API) wellKnownJwksGET(w http.ResponseWriter, _ *http.Request, _ httpr
 	api.WriteJSON(w, jwt.AccountsPublicJWKS)
 }
 
+// abortTxn is a helper method that aborts a transaction and logs a warning if
+// it fails to do it.
+func (api *API) abortTxn(sctx mongo.SessionContext) {
+	err := sctx.AbortTransaction(sctx)
+	if err != nil {
+		api.staticLogger.Warningln("Failed to abort transaction. Error:", err.Error())
+	}
+}
+
+// commitTxn is a helper method that commits a transaction and logs a warning if
+// it fails to do it.
+func (api *API) commitTxn(sctx mongo.SessionContext) {
+	err := sctx.CommitTransaction(sctx)
+	if err != nil {
+		api.staticLogger.Warningln("Failed to commit transaction. Error:", err.Error())
+	}
+}
+
+// startTxn is a helper method which starts a Mongo Session and then starts a
+// new transaction with it.
+func startTxn(ctx context.Context) (mongo.SessionContext, error) {
+	sess := mongo.SessionFromContext(ctx)
+	sctx := mongo.NewSessionContext(ctx, sess)
```

Comment:
This is also very easy to unit test since you can just add a dummy handler which you wrap with `WithDBSession` that only writes a single thing to the db and then returns an error one time and a success the other time.
So you just check whether the txn happened or not and then you can trust that as long as you pass the right context to the db, all modifications within a handler happen atomically.

Reply:
Added.